src/data/batch_generator.h

```diff
@@ -132,8 +133,14 @@ class BatchGenerator : public RNGEngine {
       if(current_ != data_->end())
         ++current_;
     }
+
+    std::deque<BatchPtr> tempBatches;
+
     size_t sets = 0;
     while(current_ != data_->end() && maxiBatch->size() < maxSize) { // loop over data
+      if (getSignalFlag(SIGTERM)) { // received SIGTERM, abandon ship ...
+        return tempBatches;
```

Comment:
If I understand that correctly that will return a set of batches that have been produced so far. Are we actually expecting that these will be processed in anyway? If not would it make more sense to maybe return an empty collection?

Reply:
Since the return value is ignored after a SIGTERM, it shouldn't matter whether it's empty or not. Even though it's marginal here, returning an empty collection involves another call to the constructor of the return value. I have no strong feelings about this.

Reply:
Now returning empty deque. It doesn't make a difference, because we need to check for SIGTERM later anyway, because SIGTERM can occur at any point, so we have to check in strategic locations, preferably in the middle of potentially long  loops (e.g., reading a million sentences from disk to create the next minibatch).